From d1bf9c52ffcf5c751d3c6ed782d29b7a8912e0a7 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Thu, 21 Oct 2021 19:37:02 -0300
Subject: [PATCH 117/233] [FIO toup] imx8mq_evk: define default dfu_alt_info
 for emmc updates

Define dfu_alt_info to allow updating SPL/imx-boot and u-boot.itb via
DFU (which is also used by EFI capsule updates).

Upstream note: standard SPL seems to be called flash.bin (generated by
u-boot), so needs to be aligned there. Here we're also using the offset
and size for what we consider the first slot (when taking multiboot as
consideration), but that shouldn't be an issue with upstream.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 include/configs/imx8mp_evk.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/imx8mp_evk.h b/include/configs/imx8mp_evk.h
index 913540988b1..61d278858f7 100644
--- a/include/configs/imx8mp_evk.h
+++ b/include/configs/imx8mp_evk.h
@@ -138,6 +138,8 @@
 	"boot_fdt=try\0" \
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fit=no\0" \
+	"dfu_alt_info=mmc 2=imx-boot raw 0x0 0x300 mmcpart 1;" \
+		"u-boot-itb raw 0x300 0xD00 mmcpart 1\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"bootm_size=0x10000000\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
-- 
2.25.1

